File: backend/src/main/java/com/dev/cinemasystem/Entity/ComboDetail.java
package com.dev.cinemasystem.Entity;

import com.dev.cinemasystem.enums.Status;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Table(
        name = "combo_detail",
        uniqueConstraints = @UniqueConstraint(
                name = "uq_combo_detail_combo_product",
                columnNames = {"combo_id", "product_id"}
        )
)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ComboDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer comboDetailId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "combo_id", nullable = false)
    Combo combo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    Product product;

    @Column(nullable = false)
    Integer quantity;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    Status status;
}

File: frontend/src/types/comboDetail.ts
import type { ComboEntity } from "./combo";
import type { ProductEntity } from "./product";

export type ComboDetailStatus = "ACTIVE" | "CANCELLED";

export type ComboDetailEntity = {
    comboDetailId: number;
    combo: ComboEntity;
    product: ProductEntity;
    quantity: number;
    status: ComboDetailStatus;
};

export type ComboDetail = ComboDetailEntity;
